lib/ruby_lsp/requests/hover.rb

```diff
@@ -54,27 +55,36 @@ def run
         when SyntaxTree::ConstPathRef
           constant_name = full_constant_name(target)
           generate_rails_document_link_hover(constant_name, target)
+        else
+          create_hover(target, "")
         end
+
+        return response if Extensions::Hover.hooks.empty?
+
+        # Run after request hooks coming from extensions and append the returned content
+        Extensions::Hover.hooks.each do |hook|
+          hook_response = hook.run(target)
+          response.contents.value << hook_response << "\n\n" unless hook_response.nil?
+        end
```

Comment:
This is not essential for this PR review, but I think this should all happen one layer above this one, i.e. at the layer where `Hover.new(document, position).run` is being called. I know that that creates a problem with how `target` is passed along, but I am sure we can find a way to handle that. For example, maybe finding the target is not the responsibility of this class in the first place.

Reply:
What's the concern with having the hooks at this level? In hover, it will be a small inconvenience because of `target`, but for other requests it may become a pretty big issue and could require us to leak a lot of request state to `Executor`.

Reply:
Again, for hover, this might be straight forward because the only requirement is `target` and there are no visits. But for requests that involve visits and require mutating a lot of state, I think it could get messy very quickly. We also want to have a different degree of control and flexibility in our own requests when compared to extensions.

Besides being a purist view, what's the benefit in following that approach?

Reply:
I think I am not sure if we are setting up any of the other hover providers for success by not treating them on the same level as we are treating our own hover request handler. But then, I think we can also ship this and see if that's true or not.